### PR TITLE
Allow footer styling

### PR DESCRIPTION
**Pull request by wilky27**
Allow the footer to be styled with css.

### CODE REVIEW

**Comment by djhvscf** [comment]
Great! :+1: @wilky27

**Comment by djhvscf** [comment]
Fix #1882

**Comment by wenzhixin** [comment]
Can you provide a jsfiddle example for the new feature, and help to update the docs?

**Comment by dabros** [comment]
@wilky27 - see https://github.com/wenzhixin/bootstrap-table/blob/master/CONTRIBUTING.md#bug-reports for fiddle bases

For doco, see how other **PR** like #1934 made seperate commit for that.

**Comment by wilky27** [comment]
jsfiddle:

http://jsfiddle.net/wilky27/9ymkp3tm/

How do I update the doco?

**Comment by dabros** [comment]
@wilky27 - see my post and the link i gave.

Same process as editing any other file, if i recall right they mostly located inside of https://github.com/wenzhixin/bootstrap-table/tree/develop/docs/_i18n, but you can go to the doco website and click 'edit in github' on most areas

http://bootstrap-table.wenzhixin.net.cn/documentation/#methods

**Comment by wilky27** [comment]
Done (i think).

Thanks.

File: src/bootstrap-table.js [hunk 1]
```diff
@@ -2002,26 +2006,37 @@
...
-            var falign = '', // footer align style
-                style = '',
+            var key,
+                falign = '', // footer align style
+                valign = '',
+                csses = [],
+                style = {},
                 class_ = sprintf(' class="%s"', column['class']);
-
+
```
Comment (Falseee) [inline src/bootstrap-table.js]: bad whitespaces :-1:

Comment (wilky27) [inline src/bootstrap-table.js]: Oh fuck off you daft cunt.

Comment (wenzhixin) [inline src/bootstrap-table.js]: Please remove the space here. The same with other lines in bottom.

**Comment by wenzhixin** [comment]
Please check out this post: https://robots.thoughtbot.com/git-interactive-rebase-squash-amend-rewriting-history. You can use like `git rebase -i HEAD~3` to squash your commits together:

```
# Rebase 8db7e8b..fa20af3 onto 8db7e8b
#
# Commands:
#  p, pick = use commit
#  r, reword = use commit, but edit the commit message
#  e, edit = use commit, but stop for amending
#  s, squash = use commit, but meld into previous commit
#  f, fixup = like "squash", but discard this commit's log message
#  x, exec = run command (the rest of the line) using shell
#
# These lines can be re-ordered; they are executed from top to bottom.
#
# If you remove a line here THAT COMMIT WILL BE LOST.
#
# However, if you remove everything, the rebase will be aborted.
#
# Note that empty commits are commented out
```

**Comment by wenzhixin** [comment]
Thanks~!

**Pull request merged**